packages/react-instantsearch-core/src/connectors/connectConfigure.js

```diff
@@ -16,37 +16,42 @@ export default createConnector({
     return {};
   },
   getSearchParameters(searchParameters, props) {
-    const items = omit(props, 'children');
+    const items = omit(props, 'children', 'contextValue');
     return searchParameters.setQueryParameters(items);
   },
   transitionState(props, prevSearchState, nextSearchState) {
     const id = getId();
-    const items = omit(props, 'children');
+    const items = omit(props, 'children', 'contextValue');
     const nonPresentKeys = this._props
-      ? difference(keys(this._props), keys(props))
+      ? difference(Object.keys(this._props), Object.keys(props))
       : [];
     this._props = props;
     const nextValue = {
       [id]: { ...omit(nextSearchState[id], nonPresentKeys), ...items },
     };
-    return refineValue(nextSearchState, nextValue, this.context);
+    return refineValue(nextSearchState, nextValue, { ais: props.contextValue });
```

Comment:
At the moment this is not correct, we should be very mindful when we review the PR on the multi-index.